src/main/java/com/iexec/resultproxy/jwt/JwtService.java

```diff
@@ -34,30 +55,45 @@ private String getOrCreateJwt(String walletAddress) {
         if (oExistingJwt.isPresent()) {
             jwtString = oExistingJwt.get().getJwtString(); // TODO generate new token
         } else {
-            jwtString = Jwts.builder().setAudience(walletAddress).setIssuedAt(new Date())
-                    .setSubject(RandomStringUtils.randomAlphanumeric(64)).compact();
+            jwtString = createJwt(walletAddress);
             save(new Jwt(walletAddress, jwtString));
         }
         return jwtString;
     }
 
-    public boolean isValidJwt(String jwtString) {
-        String claimedWalletAddress = getWalletAddressFromJwtString(jwtString);
-
-        Optional<Jwt> oExistingJwt = findByWalletAddress(claimedWalletAddress);
+    String createJwt(String walletAddress) {
+        return Jwts.builder()
+                .setAudience(walletAddress)
+                .setIssuedAt(new Date())
+                .setSubject(RandomStringUtils.randomAlphanumeric(64))
+                .signWith(SignatureAlgorithm.HS256, jwtKey)
+                .compact();
+    }
 
-        return oExistingJwt.isPresent() && jwtString.equals(oExistingJwt.get().getJwtString());
+    public boolean isValidJwt(String jwtString) {
+        try {
+            String claimedWalletAddress = getWalletAddressFromJwtString(jwtString);
```

Comment:
What does `getWalletAddressFromJwtString()` return when the signature is not valid? We need to return without querying the db if the token is not valid.

Reply:
Oh I didn't see this test `failToReadWalletAddressOnWronglySignedToken()`. Never mind.

Reply:
Exactly, I was going to reply with a pointer on the tests.
I tried several cases with unsigned tokens, badly signed tokens, ...
Feel free to add other comments if you think there are missing test cases.